en/advanced_config/ethernet_setup.md

```diff
@@ -0,0 +1,208 @@
+- `ROUTER`: The addess of the default route.
+- `DNS`: The address of the DNS server.
+
+To set the above "example" configuration using the *QGroundControl*:
+
+1. Connect the flight controller to the computer with the USB cable.
+1. Open **QGroundcontrol > Analyze Tools > MAVLink Console**
+1. Enter commands "like" the ones below into the *MAVLink Console* (to write the values to the configuration file):
+   ```
+   echo DEVICE=eth0 > /fs/microsd/net.cfg
+   echo BOOTPROTO=fallback >> /fs/microsd/net.cfg
+   echo IPADDR=192.168.0.4 >> /fs/microsd/net.cfg
+   echo NETMASK=255.255.255.0 >>/fs/microsd/net.cfg
+   echo ROUTER=192.168.0.254 >>/fs/microsd/net.cfg
+   echo DNS=192.168.0.254 >>/fs/microsd/net.cfg
+   ```
+1. Once the network configuration has been set you can disconnect the USB cable.
+1. Reboot the flight controller to apply the settings.
+
+
+### Ubuntu Ethernet Network Setup
+
+If you're using Ubuntu for your ground station (or companion computer) then you can use [netplan](https://netplan.io/) to configure the network.
+
+Below we show how you write a setup to the netplan configuration file "`/etc/netplan/01-network-manager-all.yaml`", which would run on the same network as used by the PX4 setup above.
+Note that there are many more [examples](https://netplan.io/examples/) and instructions in the [netplan](https://netplan.io/) documentation.
+
+To setup the Ubuntu Computer:
+
+1. In a terminal, create and open a `netplan` configuration file: `/etc/netplan/01-network-manager-all.yaml`
+   Below we do this using the *nano* text editor.
+
+   ```
+   sudo nano /etc/netplan/01-network-manager-all.yaml 
+   ```
+
+1. Copy and paste the following configuration information into the file (note: the indentations are important!):
+   ```
+   network:
+     version: 2
+     renderer: NetworkManager
+     ethernets:
+         enp2s0:
+             addresses:
+                 - 192.168.0.1/24
+             nameservers:
+                 addresses: [192.168.0.1]
+             routes:
+                 - to: 192.168.0.1
+                   via: 192.168.0.1
+   ```
+   
+   Save and exit the editor.
+1. Apply the *netplan* configuration by entering the following command into the Ubuntu terminal.
+   ```
+   sudo netplan apply
+   ```
+   
+If you have already configured the Pixhawk 5X, at this point you could connect it to the computer via an Ethernet cable and it would be automatically deteced by QGroundControl.
+  
+### Companion Computer Ethernet Network Setup
+
+The setup for a companion computer will depend on the companion computer operating system.
+A Linux operating system may support netplan, in which case the instructions would be the same as above, but using a unique IP address.
+
+<!-- Then link or continue on with MAVSDK setup - depending on what works when you try it -->
+
+## MAVSDK Sanity check
+
+Since we have to listen to port 14550 as mentioned earlier, the only thing changes in MAVSDK is the port we connect to UDP. 
```

Comment:
This might "work" but is not correct. You'll want to change the port and streamed mavlink messages as outlined in " PX4 Ethernet Port Configuration". The set of messages a GCS needs and an onboard computer are different.